src/site/shared/containers/ContextMenu/index.tsx

```diff
@@ -55,7 +56,28 @@ export const ContextMenu = ({info, paste}: Props) => {
             else if (ev.type === "mousedown")
                 dispatch(CloseContextMenu());
         });
-    }, [input])
+    }, [input]);
+
+    // Position changes are calculated using the react hook so that the
+    // context menu does not jump around during other update events.
+    // fixes issue #914
+    useEffect(() => {
+        if (!isOpen)
+            return;
+        // Updates position state
+        let pos = input?.getMousePos();
+        setPos({ posX:pos.x, posY:pos.y });
+
+
+    }, [isOpen]);
+
+    useDocEvent("mousedown", (ev) => {
+        if (!menu.current)
+            throw new Error("ContextMenu failed: menu.current is null");
+
+        if (!menu.current.contains(ev.target as Node))
+            dispatch(CloseContextMenu());
+    });
```

Comment:
add blank (`[]`) dependency list to this `useDocEvent` otherwise, it'll add some overhead